### ISSUE TITLE
unarchive module fails for àâæçéèïîôœ(copy)!@#$%^&-().jpg*

### ISSUE THREAD

**Issue by abi-rangaswamy** [ansible/ansible#14283]
**Ansible error output :**

fatal: [puppetclient.xxxxx.me]: FAILED! => {"changed": false, "failed": true, "msg": "Unexpected error when accessing exploded file: [Errno 2] No such file or directory: '/home/services/xxxx/xxxx/xxxxx./test/phpunit/modules/xxxx/storage/a\\314\\200a\\314\\202\\303\\246c\\314\\247e\\314\\201e\\314\\200i\\314\\210i\\314\\202o\\314\\202\\305\\223(copy)!@#$%^&-().jpg'"}

**Original file:**   àâæçéèïîôœ(copy)!@#$%^&-().jpg*

Extracting file using tar at server works fine

**Comment by jimi-c** [comment]
Hi @abishekabi, with which version of Ansible are you seeing this?

**Comment by abadger** [comment]
I'm not able to reproduce with either stable-1.9 or devel  branches at the moment.  I think this is likely due to very specific encoding and locale settings.   What is the encoding of the filename inside of the tarfile?  Are you able to give us the tarfile with just that one file in it (or an empty file that has that same filename)?  What are your locale settings?

**Comment by abadger** [comment]
In case it comes in handy for others, the error message contains the octal (how bash and other shells typically show characters that aren't in the current locale) byte values for the utf-8 encoding of the filename.

**Comment by abi-rangaswamy** [comment]
@jimi-c  ansible version
ansible 2.0.0.2
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides

**Comment by abi-rangaswamy** [comment]
@abadger My ansible control/ machine locale
root@abishek:/home/abishek# locale
LANG=en_IN
LANGUAGE=en_IN:en
LC_CTYPE="en_IN"
LC_NUMERIC="en_IN"
LC_TIME="en_IN"
LC_COLLATE="en_IN"
LC_MONETARY="en_IN"
LC_MESSAGES="en_IN"
LC_PAPER="en_IN"
LC_NAME="en_IN"
LC_ADDRESS="en_IN"
LC_TELEPHONE="en_IN"
LC_MEASUREMENT="en_IN"
LC_IDENTIFICATION="en_IN"

## LC_ALL=

Ubuntu 14.04

Unzip below file 
[testfile.zip](https://github.com/ansible/ansible/files/121254/testfile.zip)

Format: **tar.gz**

**Error:**
TASK [unarchive] ***************************************************************
fatal: [client.xxxxxx.me]: FAILED! => {"changed": false, "failed": true, "msg": "Unexpected error when accessing exploded file: [Errno 2] No such file or directory: '/home/abishek/storage/a\\314\\200a\\314\\202\\303\\246c\\314\\247e\\314\\201e\\314\\200i\\314\\210i\\314\\202o\\314\\202\\305\\223(copy)!@#$%^&-().jpg'"}

**Comment by abadger** [comment]
Working on old tickets... I was still unable to reproduce this:

```
$ LANG=en_IN LC_ALL=en_IN LANGUAGE=en_IN:en ansible-playbook -i 'ubuntu14,' 14283.yml -vv
Using /srv/ansible/troubleshooting/ansible.cfg as config file
1 plays in 14283.yml

PLAY ***************************************************************************

TASK [setup] *******************************************************************
ok: [ubuntu14]

TASK [unarchive] ***************************************************************
changed: [ubuntu14] => {"changed": true, "check_results": {"cmd": "/bin/tar -C \"/var/tmp/unarchive\" --diff -zf \"/home/badger/.ansible/tmp/ansible-tmp-1458488370.92-243333600846175/source\"", "err": "/bin/tar: storage: Warning: Cannot stat: No such file or directory\n/bin/tar: storage/àâæçéèïîôœ(copy)!@#$%^&-().jpg: Warning: Cannot stat: No such file or directory\n", "out": "", "rc": 1, "unarchived": false}, "dest": "/var/tmp/unarchive", "extract_results": {"cmd": "/bin/tar -xzf \"/home/badger/.ansible/tmp/ansible-tmp-1458488370.92-243333600846175/source\"", "err": "", "out": "", "rc": 0}, "gid": 1000, "group": "badger", "handler": "TgzArchive", "mode": "0775", "owner": "badger", "size": 4096, "src": "/home/badger/.ansible/tmp/ansible-tmp-1458488370.92-243333600846175/source", "state": "directory", "uid": 1000}

PLAY RECAP *********************************************************************
ubuntu14                   : ok=2    changed=1    unreachable=0    failed=0  
```

I tried both from Fedora 23 to ubuntu14 and from ubuntu14 to localhost.

I note that the locale setting you're showing me (en_IN) does not have an encoding specifier.  I'm not sure how ubuntu generates locale settings... perhaps without an encoding specifier, the locale is in a single byte encoding that cannot handle all unicode characters?  Try setting LANG=en_IN.UTF-8 LC_ALL=en_IN.UTF-8 on both systems and seeing if that works.

**Comment by abadger** [comment]
We think this was fixed via the following two commits: https://github.com/ansible/ansible-modules-core/commit/82aeaed45d5a283a15e8f62dd641b67d04e9876e
https://github.com/ansible/ansible-modules-core/commit/0ec3b06208a7d95006148e02efac72404257a76a

Those resolved an issue with tar and non-ascii filenames when the locale was C.  It likely affected ansible usage with any non-utf8 locale.  The fix is in the stable-2.1 and devel branches so it should be present in ansible-2.1.0rc2 when we release that.
